Mathlib/Algebra/Module/NestAlgebra.lean

```diff
@@ -0,0 +1,91 @@
+/-
+Copyright (c) 2024 Christopher Hoskin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christopher Hoskin
+-/
+
+import Mathlib.Algebra.Module.Submodule.Lattice
+import Mathlib.Order.Sublattice
+import Mathlib.Algebra.Algebra.Subalgebra.Basic
+import Mathlib.LinearAlgebra.Basis.Defs
+
+/-!
+# Nest algebras
+
+Nest algebras are an operator analogue of upper triangular matrices. They have been extensively
+studied in the context of Hilbert spaces, but also in more general contexts.
+
+This file develops the algebraic aspects of the theory.
+
+## Main definitions
+
+- `NestAlg`: The nest algebra generated by a `Nest`
+- `Basis.toNest`: Every basis of a free module generates a `Nest` (and hence a nest subalgebra)
+
+## References
+
+- [K. R. Davidson, *Nest algebras*][davidson1988]
```

Comment:
Can you add [David Larson's paper](https://www.google.com/url?sa=t&source=web&rct=j&opi=89978449&url=https://www.jstor.org/stable/1971180&ved=2ahUKEwidzMrt1tiJAxWOCnkGHWlvJREQFnoECB0QAQ&usg=AOvVaw0uePvpWRXNMsytH0wCK-lE) here?